### PR TITLE
DOC: section in indexing user guide to show use of np.where

### PR DESCRIPTION
**Pull request by suvayu**
- [ ] closes #xxxx
- [ ] tests added / passed
- [ ] passes `black pandas`
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

Add a section in the user guide to illustrate the use of `np.where` and `np.select` to enlarge a dataframe conditionally.  The example is taken from SO, as mentioned in [this issue](https://github.com/MarcoGorelli/PyDataGlobal2020-sprint/issues/5#issuecomment-727241262).

### CODE REVIEW

File: doc/source/user_guide/indexing.rst [hunk 1]
```diff
@@ -1158,6 +1158,39 @@ Mask
...
+When you have two choices to choose from, say in the following dataframe, set the color
+to 'green' when the second column has 'Z'.  You can do the following:
+
+.. ipython:: python
+
+   df = pd.DataFrame({'col1': list('ABBC'), 'col2': list('ZZXY')})
+   df['color'] = np.where(df['col2'] == 'Z', 'green', 'red')
+   df
+
+If you have more than one conditions, you can use :func:`numpy.select` to achieve that.
```
Comment (ivanovmg) [inline doc/source/user_guide/indexing.rst:1179]: ``more that one condition`` (singular)
Or maybe replace it with ``multiple conditions``?

File: doc/source/user_guide/indexing.rst [hunk 2]
```diff
@@ -1158,6 +1158,39 @@ Mask
...
+
+Setting with enlargement conditionally using :func:`numpy`
+----------------------------------------------------------
+
+An alternative to :meth:`~pandas.DataFrame.where` is to use :func:`numpy.where`.
+Combined with setting a new column, you can use it to enlarge a dataframe where the
+values are determined conditionally.
+
+When you have two choices to choose from, say in the following dataframe, set the color
+to 'green' when the second column has 'Z'.  You can do the following:
```
Comment (ivanovmg) [inline doc/source/user_guide/indexing.rst:1171]: Probably this phrasing will be slightly simpler to comprehend?

```
Consider you have two choices to choose from in the following dataframe.
To set the color to 'green' when the second column contains 'Z', you can do the following.
```

But maybe not, just a comment.

**Review by ivanovmg — COMMENTED** [review]
Looks good to me.
Nice and clear.
Couple of comments below.
Pending green.

File: doc/source/user_guide/indexing.rst (hunk 1 above)
Comment (suvayu) [reply doc/source/user_guide/indexing.rst]: done

File: doc/source/user_guide/indexing.rst (hunk 2 above)
Comment (suvayu) [reply doc/source/user_guide/indexing.rst]: I reworded it, but slightly differently.  Let me know what you think :)

Comment (ivanovmg) [reply doc/source/user_guide/indexing.rst]: Sounds good. Thank you!

**Review by jreback — APPROVED** [review]
lgtm. thanks @jd

File: doc/source/user_guide/indexing.rst [hunk 3]
```diff
@@ -1158,6 +1158,40 @@ Mask
    s.mask(s >= 0)
    df.mask(df >= 0)
 
+.. _indexing.np_where:
+
+Setting with enlargement conditionally using :func:`numpy`
+----------------------------------------------------------
+
+An alternative to :meth:`~pandas.DataFrame.where` is to use :func:`numpy.where`.
```
Comment (jorisvandenbossche) [inline doc/source/user_guide/indexing.rst:1166]: What is the difference compared to using `DataFrame.where` ?

Comment (suvayu) [reply doc/source/user_guide/indexing.rst:1166]: Hi Joris, with `np.where`, enlarging the dataframe by adding a column is straightforward.  Also, it naturally extends to multiple conditions with `np.select`.

**Comment by jreback** [comment]
thanks @suvayu

File: doc/source/user_guide/indexing.rst (hunk 3 above)
Comment (jorisvandenbossche) [reply doc/source/user_guide/indexing.rst:1166]: @suvayu thanks for the answer! Looking at the example now, I think the main difference is that with np.where, your x/y values to take from don't need to be a Series to start with (eg `Series.where` always takes the values from the calling Series where the condition is True, and so indeed you wouldn't be able to achieve the actual example here)

**Pull request merged**